### PR TITLE
[RDD cache PMem extension] [POAE7-316] Move RDD cache PMem extension into OAP project - initial part

### PR DESCRIPTION
**Pull request by winningsix**
## What changes were proposed in this pull request?

We will use script to automatically generated changed Spark to lower the effort. From now on, we only need to update the change file lists and add patch to apply. This PR is just the initial part to demonstrate how it will work.

This is for master branch. Will create another PR for 0.8 branch.

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -42,4 +42,681 @@
         <module>oap-cache/oap</module>
     </modules>
 
+  <groupId>com.intel.ssg.bdt</groupId>
```
Comment (yeyuqiang) [inline pom.xml]: duplicated groupid and artifactid. Looks like you are refer to a very old oap version

File: oap-spark/pom.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>oap</artifactId>
+        <groupId>com.intel.ssg.bdt</groupId>
```
Comment (yeyuqiang) [inline oap-spark/pom.xml]: wrong groupdid

File: oap-spark/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+https://github.com/Intel-bigdata/Spark. Please make sure all your changes are committed to the
+repository above. Otherwise, your change will be override by others.
+
+The files from this package should avoid depending on other OAP module except OAP-Common.
+
+All Spark related changes are tracked in dev/changes_list/spark_changed_files
+
+All changed files are ordered by file name.
+
+You can execute the script dev/Apply_Spark_changes.sh with the specified Spark source directories
```
Comment (yeyuqiang) [inline oap-spark/README.md:14]: Did not find this script in this PR.

**Comment by winningsix** [comment]
@zhouyuan Why do we want to ignore all sh files? I want to add some for dev purpose.

**Comment by yeyuqiang** [comment]
Overall LGTM

**Pull request merged**